neurow/lib/neurow/internal_api/endpoint.ex

```diff
@@ -53,21 +56,31 @@ defmodule Neurow.InternalApi do
 
   post "/v1/publish" do
     case extract_params(conn) do
-      {:ok, message, topic} ->
-        message_id = to_string(:os.system_time(:millisecond))
+      {:ok, messages, topics} ->
+        publish_timestamp = :os.system_time(:millisecond)
+
+        nb_publish = length(messages) * length(topics)
 
-        :ok =
-          Phoenix.PubSub.broadcast!(Neurow.PubSub, topic, {:pubsub_message, message_id, message})
+        Enum.each(topics, fn topic ->
+          Enum.each(messages, fn message ->
+            Phoenix.PubSub.broadcast!(
+              Neurow.PubSub,
+              topic,
+              {:pubsub_message,
+               %Message{message | timestamp: message.timestamp || publish_timestamp}}
+            )
 
-        Logger.debug("Message published on topic: #{topic}")
-        Stats.inc_msg_received()
+            Stats.inc_msg_received()
```

Comment:
Not sure what we really want to monitor with the new API.

Reply:
Should we incr with nb_publish?

Reply:
Do we want to count the number of call on the internal interface per issuer ?

Reply:
In that case, it should be outside of the loop :)